tests/geolocation/test_geolocation_shared_via_w3c_api.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+from selenium.common import NoSuchElementException
+from selenium.webdriver import Firefox
+from selenium.webdriver.common.by import By
+
+from modules.browser_object import Navigation, TabBar
+from modules.page_object import GenericPage
+
+
+@pytest.fixture()
+def test_case():
+    return "15186"
+
+
+@pytest.fixture()
+def set_prefs():
+    """Set prefs"""
+    return [
+        (
+            "geo.provider.network.url",
+            "https://www.googleapis.com/geolocation/v1/geolocate?key"
+            "=%GOOGLE_LOCATION_SERVICE_API_KEY%",
+        )
+    ]
+
+
+TEST_URL = "https://www.w3schools.com/html/html5_geolocation.asp"
+
+
+def handle_cookie_banner(driver, web_page):
+    """
+    Address the cookie banner manually if appears, as the cookie banner dismissal preference is not effective in this
+    context
+    """
+    try:
+        driver.switch_to.window(driver.window_handles[-1])
+        web_page.find_element(By.ID, "accept-choices").click()
+    except NoSuchElementException:
+        # If the cookie banner is not found, continue with the test
+        pass
+
+
+def handle_geolocation_prompt(nav, button_type="primary"):
```

Comment:
Could you move this to the navigation browser object? I think this could be useful in other tests in the future...